wled00/FX.cpp

```diff
@@ -2032,7 +2032,7 @@ uint16_t mode_palette() {
       const mathType sourceX = xtSinTheta + ytCosTheta + centerX;
       // The computation was scaled just right so that the result should always be in range [0, maxXOut], but enforce this anyway
       // to account for imprecision. Then scale it so that the range is [0, 255], which we can use with the palette.
-      int colorIndex = (std::min(std::max(sourceX, mathType(0)), maxXOut * sInt16Scale) * 255) / (sInt16Scale * maxXOut);
+      int colorIndex = (std::min(std::max(sourceX, mathType(0)), maxXOut * sInt16Scale) * wideMathType(255)) / (sInt16Scale * maxXOut);
```

Comment:
What does `wideMathType()` do? 

Essentially this PR changes "255" to "wideMathType(255)", I'm wondering is this the same as writing "255.0f" or maybe "255UL" ?

Reply:
255ULL for ESP8266 and 255.0f otherwise

Reply:
Looks like someone tried to avoid fixed point math?

Reply:
the way it is implemented is quite an elegant way to support float AND fixed point math.